### PR TITLE
Test setup

### LINKED ISSUE #2: Testing: npm link .. fails for me
```
npm link ..
npm ERR! Darwin 15.6.0
npm ERR! argv "/Users/gregor/.nvm/versions/node/v6.4.0/bin/node" "/Users/gregor/.nvm/versions/node/v6.4.0/bin/npm" "link" ".."
npm ERR! node v6.4.0
npm ERR! npm  v3.10.3

npm ERR! May not delete: /Users/gregor/Projects/pouchdb-adapter-fs/pouchdb
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/gregor/Projects/pouchdb-adapter-fs/pouchdb/npm-debug.log
```

I suggest we add a section `Testing` to the README and ask people to git clone `pouchdb` into a different repository, install dependencies and then run `npm link /path/to/pouchdb-adapter-fs` and then run `CLIENT_ADAPTER=pouchdb-adapter-fs npm test` in the `pouchdb` folder?

That would be a simple first step. In general it would be nice to have a best practice for PouchDB adapter developers to test their adapter using the pouchdb test suite

### PR DESCRIPTION
**Pull request by gr2m**
closes #2

I copied all integration tests from PouchDB@5.4.5 and made adjustments to make them run. I had to skip a few tests, it would be great if we could review them.
The test setup is not ideal as we’d need to keep it updated manually with new PouchDB versions, but it’s a good start?

Ideally PouchDB’s monorepo would release an integration test script with each PouchDB release that plugin & adapter developers can use to test their creations :)

### CODE REVIEW

**Comment by nolanlawson** [comment]
I'm fine with copy-pasta'ing tests for now. Every project I've ever worked on that tried to "share" tests (e.g. LevelDOWN & friends) eventually became an unmaintainable mess of environment variables, commented-out tests, etc. etc.

FWIW this is also the strategy I did for worker-pouch and socket-pouch. Far too difficult to try to share tests; copy-pasta tastes so good in comparison! 🍝

**Comment by nolanlawson** [comment]
So yeah :+1:

**Comment by gr2m** [comment]
Can you please active this repo on Travis? Or make me admin (maybe for both this repo and fsdown)?

**Comment by nolanlawson** [comment]
All the `oldStyle`/`newStyle` stuff BTW is only needed for `test.migration.js`, which tests LevelDB migrations. Not necessary for this project.

**Comment by nolanlawson** [comment]
Actually @gr2m you can just remove migration, prefix, issue9**, and http test files. All those tests don't really apply to custom adapters.

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
-script: npm test
+env:
+  matrix:
+  - COUCH_HOST=http://localhost:5984
 branches:
-  only:
-  - master
-  - /^greenkeeper-.*$/
+  except:
+    - /^v\d+\.\d+\.\d+$/
```
Comment (gr2m) [inline .travis.yml:14]: @nolanlawson is it okay with you to run builds for everything but tags?

Comment (nolanlawson) [reply .travis.yml:14]: Yup, I'm fine with that

**Pull request merged**